esp-hal-common/src/timer.rs

```diff
@@ -19,6 +20,7 @@ pub enum Error {
 /// General-purpose timer
 pub struct Timer<T> {
     timg: T,
+    apb_clk_freq: MegahertzU32,
```

Comment:
Should we take both `APB_CLK` and `XTAL_CLK` frequencies here, or should we expect the user to provide a single frequency value to use, based on the peripheral configuration?

Reply:
I think it's fine for now. We could assert that `APB_CLOCK` is configured but since we don't offer an API to do that I guess someone who is using the PAC or raw register writes to change it should be aware of the consequences

Sure once we offer an API to change this we should reconsider this